kroxylicious-schema-tools/kroxylicious-maven-plugin/src/main/java/io/kroxylicious/CompilePluginMojo.java

```diff
@@ -0,0 +1,79 @@
+    @Override
+    protected SchemaCompiler schemaCompiler() throws MojoExecutionException {
+        return new SchemaCompiler(List.of(source.toPath()),
+                null,
+                readHeaderFile(),
+                existingClasses != null ? existingClasses : Map.of(),
+                List.of(),
+                List.of(
+                        new PropertyAnnotator() {
+                            @Override
+                            public List<AnnotationExpr> annotateConstructorParameter(
+                                                                                     Diagnostics diagnostics,
+                                                                                     String property,
+                                                                                     SchemaObject propertySchema) {
+
+                                if (List.of(SchemaType.STRING).equals(propertySchema.getType())
+                                        && "plugin-impl-name".equals(propertySchema.getFormat())) {
+                                    Object o = propertySchema.getAdditionalProperties().get("plugin-interface-name");
+                                    if (o instanceof String interfaceName) {
+                                        return List.of(new SingleMemberAnnotationExpr(
+                                                new Name("io.kroxylicious.proxy.plugin.PluginImplName"),
+                                                new ClassExpr(new ClassOrInterfaceType(null,
+                                                        interfaceName))));
+                                    }
+                                    diagnostics.reportError("`format: plugin-impl-name` requires the `plugin-interface-name` property");
+                                }
+                                else if (List.of(SchemaType.OBJECT).equals(propertySchema.getType())
+                                        && "plugin-impl-config".equals(propertySchema.getFormat())) {
+                                    Object o = propertySchema.getAdditionalProperties().get("impl-name-property");
+                                    if (o instanceof String implName) {
+                                        return List.of(new NormalAnnotationExpr(
+                                                new Name("io.kroxylicious.proxy.plugin.PluginImplConfig"),
+                                                NodeList.nodeList(
+                                                        new MemberValuePair(new SimpleName("implNameProperty"),
+                                                                new StringLiteralExpr(implName)))));
+                                    }
+
+                                    diagnostics.reportError("`format: plugin-impl-config` requires the `impl-name-property` property");
+                                }
+
+                                return List.of();
```

Comment:
We don't generate any errors if there is a field named `plugin-impl-name` that has a non string type. Similarly if `plugin-impl-config` was a simple scalar value it would be ignored (I think) and no errors reported.

Am I miss reading that? Should we report those errors?